### PR TITLE
#84 messaging

### PR DESCRIPTION
**Pull request by neo7337**

### CODE REVIEW

**Comment by neo7337** [comment]
TODO::
1. test cases
2. local implementation testing

File: clients/messaging/message.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	// ReadContent reads the content body based on the contentType from the Message structure
+	ReadContent(out interface{}, contentType string) error
+}
+
+// Message interface wil be implemented by all third party implementation such as aws - sns, sqs, gcp -> pub/sub, gcm, messaging -> amqp, kafka
+type Message interface {
+	Header
+	Body
+	// TODO :: do we need this?
+	io.ReadWriter
```
[the remark this answers is not in the document]
Comment (nandagopalan) [reply clients/messaging/message.go:82]: we can remove this one. The concrete implementations may never use this as they are lib specific for that provider

**Comment by neo7337** [comment]
@nandagopalan have a look at the new archi
A couple of TODO questions, please have a look at them!

**Review by nandagopalan — APPROVED** [review]
Merge to main

**Pull request merged**